src/MergeContext.js

```diff
@@ -454,6 +454,222 @@ class BranchPosition
+    // authorField is a {name, value, raw}
+    _parseAuthor(authorField) {
+        const trimmedValue = this._trim(authorField.value);
+        const cred = trimmedValue.match(/^ ([\w][^@<>,]*) <(\S+@\S+\.\S+)>$/);
+        if (!cred)
+            throw new Error(`unsupported ${authorField.name} value format: ${authorField.value}`);
+
+        if (cred[0].includes(','))
+            throw new Error(`${authorField.name} author name with a comma: ${authorField.value}`);
+
+        return {name: cred[1].trim(), email: cred[2].trim()};
+    }
+
+    _extractHeader(prDescriptionRaw) {
+        const prDescription = this._trim(prDescriptionRaw);
+        if (prDescription.startsWith('Authored-by:')) {
+            let tokenizer = new FieldsTokenizer(prDescription);
+            const authorField = tokenizer.nextField();
+            this._customAuthor = this._parseAuthor(authorField);
+            if (!tokenizer.atEnd())
+                throw new Error(`unexpected header lines after a single Authored-by attribute`);
+            return tokenizer.remaining();
+        } else {
+            return prDescription;
+        }
+    }
+
+    _extractTrailer(prDescriptionWithoutHeaderRaw) {
+        const prDescriptionWithoutHeader = this._trim(prDescriptionWithoutHeaderRaw);
+
+        // index of the last occurrence of '\n\n'
+        const trailerIndex = prDescriptionWithoutHeader.search(/\n\n(?!.*\n\n)/s);
+        if (trailerIndex < 0) // the text has at most one paragraph
+            trailerIndex = 0;
+
+        this._parseTrailer(prDescriptionWithoutHeader.substring(trailerIndex));
+        return prDescriptionWithoutHeader.substring(0, trailerIndex);
+    }
+
+    _parseBody(prDescriptionWithoutHeaderAndTrailerRaw) {
+        const prDescriptionWithoutHeaderAndTrailer = this._trim(prDescriptionWithoutHeaderAndTrailerRaw);
+        if (prDescriptionWithoutHeaderAndTrailer.length > 0) {
+            this._checkForTypos(prDescriptionWithoutHeaderAndTrailer);
+            this._body = prDescriptionWithoutHeaderAndTrailer;
+        }
+    }
+
+    _parseTrailer(trailerRaw) {
+        const trailer = this._trim(trailerRaw);
+        let tokenizer = new FieldsTokenizer(trailer, false);
+
+        if (tokenizer.atEnd())
+            throw new Error(`an empty trailer`);
+
+        while (!tokenizer.atEnd()) {
+            const field = tokenizer.nextField();
+            if (field.name === "Co-authored-by") {
+                const coAuthor = JSON.stringify(this._parseAuthor(field));
+                this._log(`accepting trailer field: ${coAuthor}`);
+            } else {
+                this._checkForTypos(field.raw);
+            }
+        }
+
+        if (trailer.length > 0)
+            this._trailer = trailer;
+    }
+
+    // checks the PR message (or its part) for some common/expected typos that may occur
+    // when filling in PR attributes
+    _checkForTypos(text) {
+        const possibleAuthoredByTypoRegex = /^\s*\S*[aA]uthored[-_]?[bB]y/m;
```

Comment:
Let's make it even more strict by making the whole pattern case-insensitive. We can relax it later if needed.

```suggestion
        const possibleAuthoredByTypoRegex = /^\s*\S*authored[-_]?by/mi;
```

Reply:
Ok.